hack/tools/kapinger/clients/http_service_controller.go

```diff
@@ -0,0 +1,111 @@
+package clients
+
+import (
+	"context"
+	"fmt"
+	"slices"
+	"sync"
+	"time"
+
+	"log"
+
+	"golang.org/x/exp/rand"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+)
+
+type serviceController struct {
+	sync.RWMutex
+	serviceInformer cache.SharedIndexInformer
+
+	ips []string
```

Comment:
Thoughts on using net.IP instead of string? What stops me from putting 'wingding' as a value?

Reply:
no complaints, updated

but also if service.Spec.ClusterIP changes from string type, I think the world might stop :)